src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,121 @@
+        ) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+
+            int affectedRows = statement.executeUpdate();
+            if (affectedRows < 1) {
+                throw new RuntimeException("Expected to insert at least one row but inserted: "
+                        + affectedRows);
+            }
+
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                book.setId(generatedKeys.getObject(1, Long.class));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't save a book: " + book, e);
+        }
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        String query = "SELECT * FROM books WHERE ID = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                var statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                String title = resultSet.getString(TITLE);
+                BigDecimal price = resultSet.getBigDecimal(PRICE);
+
+                return Optional.of(new Book(id, title, price));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't find a book with id " + id, e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Book> findAll() {
+        List<Book> books = new ArrayList<>();
+
+        String query = "SELECT * FROM books";
+        try (Connection connection = ConnectionUtil.getConnection();
+                var statement = connection.prepareStatement(query)) {
+
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                long id = resultSet.getLong(ID);
+                String title = resultSet.getString(TITLE);
+                BigDecimal price = resultSet.getBigDecimal(PRICE);
+                books.add(new Book(id, title, price));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Error while finding all books", e);
+        }
+        return books;
+    }
+
+    @Override
+    public Book update(Book book) {
+        String query = "UPDATE books SET title = ? , price = ? WHERE ID = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                var statement = connection.prepareStatement(query)) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            statement.setLong(3, book.getId());
+
+            int updatedRows = statement.executeUpdate();
+            if (updatedRows < 1) {
+                throw new RuntimeException("Expected to update at least one row but updated: "
+                        + updatedRows);
```

Comment:
```suggestion
                throw new DataProcessingException("Expected to update at least one row but updated: "
                        + updatedRows);
```
I belive we can throw our custom exception here also.